File: Medii de proiectare si programare/java/lab5/ChatServices/src/main/java/com/example/IService.java
package com.example;

import java.time.LocalDate;
import java.util.List;

public interface IService {

    void logInEmployee(String email, String password, IServiceObserver client) throws Exception;

    void logOut(EmployeeAtOffice employeeAtOffice) throws Exception;

    List<Show> searchArtistByDate(LocalDate date);

    List<Show> getAllShows(EmployeeAtOffice employeeAtOffice) throws ChatException;

    void sellTickets(String showName, String buyerName, Integer noOFTickets) throws Exception;

    Show findOneShow(String showName);

    void updateShow(Show show) throws Exception;

    EmployeeAtOffice findEmployeeByEmailAndPassword(String email, String password) throws Exception;

}

File: Medii de proiectare si programare/C#/tema7/service/IService.cs
using System;
using System.Collections.Generic;
using model.domains;

namespace service
{
    public interface IService
    {
        void LogIn(string email, string password, IServiceObserver client);
        void LogOut(EmployeeAtOffice employeeAtOffice, IServiceObserver client);
        List<Show> SearchArtistByDate(DateTime date);
        List<Show> GetAllShows(EmployeeAtOffice employeeAtOffice);
        void SellTickets(string showName, string buyerName, int noOfTickets);
        Show findOneShow(string showName);
        Show UpdateShow(Show show);
        EmployeeAtOffice FindEmployeeByEmailAndPassword(string email, string password);
    }
}